### PR TITLE
Updates/review of crawl list merging

### PR DESCRIPTION
**Pull request by dzeber**
I have reviewed the code as requested in c40c409. R+ to the methodology; I'm proposing some minor changes to the code.

### CODE REVIEW

File: list_merge.py [hunk 1]
```diff
@@ -1,61 +1,71 @@
...
+            f.write(tld + "\n")
 
+
+# Holder for the combined list
+final_list = []
+
+# Open and parse the ALEXA TLD list from local csv.
+a = read_list_csv(ALEXA_LIST_FILE)
 # Truncate ALEXA list to minimum necessary elements to process.
-a = a[0 : FINAL_IDX + 1]
```
Comment (dzeber) [inline list_merge.py:20]: This looks like an off-by-1 error.

File: list_merge.py [hunk 2]
```diff
@@ -1,61 +1,71 @@
...
-with open(
-    "/Users/mlopatka/Downloads/lists/TRANCO-cannonical-top1m_24-07-2019.csv", "rb"
-) as f:
-    for line in f:
-        t.append(str(line, "UTF-8").rstrip().split(",")[1])
-
-# Truncate TANCO list to minimum necessary elements to process.
-t = t[0 : FINAL_IDX + 1]
-# Save the TANCO list into a set type for sanity checking list combination operations.
+alexa_subset_length = len(a)
```
Comment (dzeber) [inline list_merge.py:40]: Retain the length of the actual subsets for final verification.

File: list_merge.py [hunk 3]
```diff
@@ -1,61 +1,71 @@
...
+
+
+def write_list_file(tld_list, list_file):
+    """Write a top site list to file.
+
+    Writes one entry per line with no index.
+    """
+    with open(list_file, "w") as f:
+        for tld in tld_list:
+            f.write(tld + "\n")
```
Comment (dzeber) [inline list_merge.py:28]: Since this is not really a CSV, probably easiest to just do it this way.

File: list_merge.py [hunk 4]
```diff
@@ -1,61 +1,71 @@
...
+# Save the TRANCO list into a set type for sanity checking list combination operations.
 tranco_set = set(t)
+tranco_subset_length = len(t)
 
-for i in range(0, len(alexa_set)):
-    # When we have depleted the Alexa list, stop appending TRANCO elements.
-    if len(a) == 0:
-        print(len(a))
-        print(len(t))
-        break
```
Comment (dzeber) [inline list_merge.py:43]: I think we just want to iterate over the subset `a` directly. The difference between `a` and `alexa_set` will only matter if `a` has repeated elements, and in this case the final list will only retain 1 copy.

**Review by mlopatka — APPROVED** [review]
Looks good. thanks for the PR.

**Pull request merged**